orbit/pkg/update/notifications.go

```diff
@@ -352,6 +355,13 @@ func (h *runScriptsConfigReceiver) Run(cfg *fleet.OrbitConfig) error {
 		timeout = time.Duration(cfg.ScriptExeTimeout) * time.Second
 	}
 
+	if runtime.GOOS == "darwin" {
+		if cfg.Notifications.RunSetupExperience == true && !CanRun(h.rootDirPath, "swiftDialog", SwiftDialogMacOSTarget) {
```

Comment:
Ah I think I understand why those other runners also have this check, it's because the setup experience will go through them to run installs and scripts, and it cannot proceed until the swift dialog can be shown.

It won't start before the swift dialog tells them to, right? I.e. the software install/script execution requests won't be created until the swift dialog sees that profiles were delivered and all that other pre-setup experience stuff?

Reply:
> it's because the setup experience will go through them to run installs and scripts, and it cannot proceed until the swift dialog can be shown.

hey yeah this is it! swiftDialog itself will just be the UI element of the setup experience, it won't coordinate anything itself. The actual work is done by the scripts and installers and setup experiencer config receivers.